app/Http/Controllers/Frontend/Admin/CheckinController.php

```diff
@@ -0,0 +1,207 @@
+
+use App\Enum\Business;
+use App\Enum\StatusVisibility;
+use App\Enum\TravelType;
+use App\Exceptions\CheckInCollisionException;
+use App\Exceptions\HafasException;
+use App\Exceptions\StationNotOnTripException;
+use App\Exceptions\TrainCheckinAlreadyExistException;
+use App\Http\Controllers\Backend\EventController as EventBackend;
+use App\Http\Controllers\Backend\Social\MastodonController;
+use App\Http\Controllers\Backend\Social\TwitterController;
+use App\Http\Controllers\Backend\Transport\TrainCheckinController;
+use App\Http\Controllers\HafasController;
+use App\Http\Controllers\StatusController as StatusBackend;
+use App\Http\Controllers\TransportController as TransportBackend;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\Rules\Enum;
+use Illuminate\View\View;
+
+class CheckinController
+{
+
+    public function renderStationboard(Request $request): View|RedirectResponse {
+        $validated = $request->validate([
+                                            'station' => ['nullable'],
+                                            'when'    => ['nullable', 'date'],
+                                            'filter'  => ['nullable', new Enum(TravelType::class)],
+                                            'user'    => ['nullable', 'numeric']
+                                        ]);
+
+        $user = Auth::user();
+        if (isset($request->user)) {
+            $user = User::findOrFail($request->user);
+        }
+
+        $when = isset($validated['when']) ? Carbon::parse($validated['when']) : Carbon::now();
+
+        $station = $departures = $times = null;
+
+        if (isset($validated['station'])) {
+            try {
+                $TrainStationboardResponse = TransportBackend::getDepartures(
+                    stationQuery: $validated['station'],
+                    when:         $when,
+                    travelType:   TravelType::tryFrom($validated['filter'] ?? null),
+                );
+
+                $station    = $TrainStationboardResponse['station'];
+                $departures = $TrainStationboardResponse['departures'];
+                $times      = $TrainStationboardResponse['times'];
+            } catch (HafasException $exception) {
+                return back()->with('error', $exception->getMessage());
+            } catch (ModelNotFoundException) {
+                return redirect()->back()->with('error', __('controller.transport.no-station-found'));
+            }
+        }
+
+        return view('admin.checkin.stationboard', [
+            'station'    => $station,
+            'departures' => $departures,
+            'times'      => $times,
+            'when'       => $when,
+            'user'       => $user
+        ]);
+    }
+
+    public function renderTrip(string $id, Request $request) {
```

Comment:
Maybe this so we can differ?
```suggestion
    public function renderTrip(string $tripId, Request $request) {
```